### PR TITLE
Add daily challenge stats

### PR DESCRIPTION
**Pull request by nanaya**
Tried writing tests but it looks kinda horrible and I'm not quite sure if there's better way to do it ಠ_ಠ

depends on
- [x] #11350
- [x] #11343

### CODE REVIEW

File: app/Models/DailyChallengeUserStats.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        'last_update' => 'datetime',
+        'last_weekly_streak' => 'datetime',
+    ];
+    protected $primaryKey = 'user_id';
+    protected $table = 'daily_challenge_user_stats';
+
+    public static function calculate(CarbonImmutable $date): void
+    {
+        $startTime = $date->startOfDay();
+        $endTime = $startTime->endOfDay();
```
Comment (notbakaneko) [inline app/Models/DailyChallengeUserStats.php:39]: unused

File: resources/css/bem/qtip.less [hunk 2]
```diff
@@ -16,6 +18,18 @@
     }
   }
 
+  &--daily-challenge {
+    max-width: none;
+    width: 300px;
```
Comment (notbakaneko) [inline resources/css/bem/qtip.less:23]: `min-width`?

**Review by notbakaneko — CHANGES_REQUESTED** [review]
Still checking

File: database/migrations/2024_07_09_074020_create_daily_challenge_user_stats.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('daily_challenge_user_stats', function (Blueprint $table) {
+            $table->unsignedInteger('user_id')->primary();
+            $table->unsignedMediumInteger('playcount')->default(0);
```
[the remark this answers is not in the document]
Comment (nanaya) [reply database/migrations/2024_07_09_074020_create_daily_challenge_user_stats.php:18]: 🤷‍♂️

File: app/Models/Multiplayer/Room.php [hunk 4]
```diff
@@ -219,6 +223,15 @@ public function currentPlaylistItem()
...
     }
 
+    public function macroDailyChallengeFor(): \Closure
+    {
+        return fn (Builder $query, CarbonImmutable $date): ?static
+            => static::where('category', 'daily_challenge')
+                ->whereBetween('starts_at', [$date->startOfDay(), $date->endOfDay()])
+                ->orderBy('starts_at', 'DESC')
+                ->first();
+    }
```
[the remark this answers is not in the document]
Comment (nanaya) [reply app/Models/Multiplayer/Room.php:233]: maybe after a few years...

File: tests/Models/DailyChallengeUserStatsTest.php [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    {
+        return CarbonImmutable::now()->endOfWeek(CarbonImmutable::MONDAY);
+    }
+
+    private static function preparePlaylistItem(CarbonImmutable $playTime): PlaylistItem
+    {
+        return PlaylistItem::factory()->create([
+            'room_id' => Room::factory()->create([
+                'category' => 'daily_challenge',
+                'starts_at' => $playTime->startOfDay(),
```
[the remark this answers is not in the document]
Comment (nanaya) [reply tests/Models/DailyChallengeUserStatsTest.php:36]: it reads better and has fewer assumptions

**Comment by notbakaneko** [comment]
Doing the initial `daily-challenge:user-stats-calculate` and running it again counts as 2 days

**Review by notbakaneko — APPROVED** [review]
Seems to work

**Pull request merged**